### PR TITLE
feat(icon): add weightless prop

### PR DESCRIPTION
**Pull request by torgeadelin**
# Description
Added `weightless` prop. See storybook docs for more info.

# Links
[SPARK-380868](https://jira-eng-gpk2.cisco.com/jira/browse/)

### CODE REVIEW

File: src/components/Icon/Icon.stories.docs.mdx [hunk 1]
```diff
@@ -1 +1,38 @@
+import Icon from './Icon.tsx';
+
 `<Icon />` is a generic icon component that acts as an dynamic icon loader.
+
+# `weight` vs `weightless`
```
Comment (InteractiveTimmy) [inline src/components/Icon/Icon.stories.docs.mdx:5]: I like this extended description!

File: src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -28,9 +29,11 @@ const Icon: React.FC<Props> = (props: Props) => {
     style,
     title,
     weight,
+    weightless = DEFAULTS.WEIGHTLESS,
     ...otherProps
   } = props;
-  const { SvgIcon, error } = useDynamicSVGImport(`${name}-${weight || DEFAULTS.WEIGHT}`);
+  const resolvedSVGName = getResolvedSVGName(name, weight, weightless);
```
Comment (InteractiveTimmy) [inline src/components/Icon/Icon.tsx:35]: Great abstraction!

File: src/components/Icon/Icon.utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    ['icon-name', 'thin', false, 'icon-name-thin'],
+    ['icon-name', 'filled', false, 'icon-name-filled'],
+    ['icon-name', undefined, false, 'icon-name-regular'],
+    ['icon-name', undefined, true, 'icon-name'],
+    ['icon-name', 'regular', true, 'icon-name'],
+    ['icon-name', 'bold', true, 'icon-name'],
+    ['icon-name', 'thin', true, 'icon-name'],
+    ['icon-name', 'filled', true, 'icon-name'],
+  ])('returns the correct name for %s, %s, %s', (name, weight, weightless, expected) => {
+    expect(getResolvedSVGName(name, weight as IconWeight, weightless)).toEqual(expected);
```
Comment (InteractiveTimmy) [inline src/components/Icon/Icon.utils.test.ts:17]: not a fan of using iterators in tests, but, it works!

File: src/components/Icon/Icon.utils.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { IconWeight } from './Icon.types';
```
Comment (InteractiveTimmy) [inline src/components/Icon/Icon.utils.ts:1]: Should this be:

```ts
import type { IconWeight} from './Icon.types';
```

?

File: src/components/Icon/Icon.utils.test.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { IconWeight } from '.';
```
Comment (InteractiveTimmy) [inline src/components/Icon/Icon.utils.test.ts:1]: Should this be:

```ts
import type { IconWeight} from './Icon.types';
```

?

**Review by InteractiveTimmy — APPROVED** [review]
LGTM!

A few comments around styling and syntax, but nothing blocking

File: src/components/Icon/Icon.stories.args.ts [hunk 6]
```diff
@@ -98,4 +98,17 @@ export default {
       },
     },
   },
+  weightless: {
+    defaultValue: CONSTANTS.DEFAULTS.WEIGHTLESS,
+    description: 'Identifies the icons that do not have a weight.',
```
Comment (cscheffauer) [inline src/components/Icon/Icon.stories.args.ts:103]: `Indicates if the icon doesn't have a weight` would make more sense here I think

File: src/components/Icon/Icon.stories.docs.mdx [hunk 7]
```diff
@@ -1 +1,38 @@
...
+4 possible options: (thin, regular, bold, filled). Most icons from the icons library will have a
+weight variant (remark: some might only have a subset of those), but some icons will not have a
+weight variant (these are special icons): examples are brand icons, and icons like
+`webex-assistant-active`.
+
+If you're trying to use an icon which has a variant, use the `weight` prop. (Note that it defaults
+to `regular` if not provided.) If you're trying to use an icon that doesn't support weights, you'll
+have to use the `weightless` prop so that the `<Icon/>` component can find the correct .svg file.
+
+### How SVG name resolving works
```
Comment (cscheffauer) [inline src/components/Icon/Icon.stories.docs.mdx:19]: praise: awesome docs! 💯

File: src/components/Icon/Icon.utils.test.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import { getResolvedSVGName } from './Icon.utils';
+
+describe('getResolvedSVGName', () => {
+  it.each([
+    ['icon-name', 'regular', false, 'icon-name-regular'],
```
Comment (cscheffauer) [inline src/components/Icon/Icon.utils.test.ts:6]: praise: test coverage! 💯

**Review by cscheffauer — APPROVED** [review]

File: src/components/Icon/Icon.stories.args.ts (hunk 6 above)
Comment (torgeadelin) [reply src/components/Icon/Icon.stories.args.ts:103]: resolved

File: src/components/Icon/Icon.utils.ts (hunk 4 above)
Comment (torgeadelin) [reply src/components/Icon/Icon.utils.ts:1]: resolved

File: src/components/Icon/Icon.utils.test.ts (hunk 5 above)
Comment (torgeadelin) [reply src/components/Icon/Icon.utils.test.ts:1]: resolved

**Pull request merged**